src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1546,6 +1557,56 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             responder.SetResponse(response);
         }
 
+        private static long FileTimeToPosix(FILETIME ft)
+        {
+            long date = ((long)ft.dwHighDateTime << 32) + ft.dwLowDateTime;
+            // removes the diff between 1970 and 1601
+            // 100-nanoseconds = milliseconds * 10000
+            date -= 11644473600000L * 10000;
+
+            // converts back from 100-nanoseconds to seconds
+            return date / 10000000;
+        }
+
+        protected override void HandleModulesRequestAsync(IRequestResponder<ModulesArguments, ModulesResponse> responder)
+        {
+            var response = new ModulesResponse();
+
+            IEnumDebugModules2 enumDebugModules;
+            if (m_program.EnumModules(out enumDebugModules) == HRConstants.S_OK)
+            {
+                var debugModules = new IDebugModule2[1];
+
+                uint numReturned = 0;
+                while (enumDebugModules.Next(1, debugModules, ref numReturned) == HRConstants.S_OK && numReturned == 1)
+                {
+                    var debugModuleInfos = new MODULE_INFO[1];
+                    if (debugModules[0].GetInfo(enum_MODULE_INFO_FIELDS.MIF_ALLFIELDS, debugModuleInfos) == HRConstants.S_OK)
+                    {
+                        var debugModuleInfo = debugModuleInfos[0];
+                        var mod = new ProtocolMessages.Module(debugModuleInfo.m_bstrName, debugModuleInfo.m_bstrName)
+                        {
+                            Path = debugModuleInfo.m_bstrUrl,
+                            VsTimestampUTC = FileTimeToPosix(debugModuleInfo.m_TimeStamp).ToString(CultureInfo.InvariantCulture),
+                            Version = debugModuleInfo.m_bstrVersion,
+                            VsLoadAddress = debugModuleInfo.m_addrLoadAddress.ToString(CultureInfo.InvariantCulture),
+                            VsPreferredLoadAddress = debugModuleInfo.m_addrPreferredLoadAddress.ToString(CultureInfo.InvariantCulture),
+                            VsModuleSize = (int)debugModuleInfo.m_dwSize,
+                            VsLoadOrder = (int)debugModuleInfo.m_dwLoadOrder,
+                            SymbolFilePath = debugModuleInfo.m_bstrUrlSymbolLocation,
+                            SymbolStatus = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_SYMBOLS) == 0 ? "✓" : "✗",
+                            VsIs64Bit = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_64BIT) == 0,
+                            IsOptimized = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_OPTIMIZED) == 0, // not set by gdb
```

Comment:
>IsOptimized [](start = 28, length = 11)

This is more complicated than just checking a single flag because `null` is also a possible value (and what you should wind up with in the GDB case). This is true for both `IsOptimized` and `IsUserCode`.